File: app/modules/runtime/schemas.py
from __future__ import annotations

from datetime import datetime

from app.modules.shared.schemas import DashboardModel


class RuntimeVersionResponse(DashboardModel):
    current_version: str
    latest_version: str | None = None
    update_available: bool = False
    checked_at: datetime
    source: str | None = None
    release_url: str = "https://github.com/Soju06/codex-lb/releases/latest"

File: frontend/src/features/runtime/schemas.ts
import { z } from "zod";

export const RuntimeVersionSchema = z.object({
  currentVersion: z.string(),
  latestVersion: z.string().nullable().optional(),
  updateAvailable: z.boolean(),
  checkedAt: z.string(),
  source: z.string().nullable().optional(),
  releaseUrl: z.url(),
});

export type RuntimeVersion = z.infer<typeof RuntimeVersionSchema>;
